### PR TITLE
Refactor: bump hashicups, move ECS/EKS workers to private subnets

### PR DESCRIPTION
**Pull request by jjtimmons**
- bump the hashicups docker versions
- because nginx was removed in `hashicups/frontent:v1`:
  - add application load balancer to ECS
  - add nginx to EC2
- move intentions to `/modules` (ie out of the templates)
- move EKS workers to private subnets, update template variables

### Testing

For each setup, ran `tf apply` and visited the hashicups URL. Placed an order. Made sure no stderr in the browser console.

For the EC2 setup, checked the nomad UI

Screen Shot 2022-03-25 at 12 01 09 PM

Screen Shot 2022-03-25 at 1 38 15 PM

### CODE REVIEW

File: modules/hcp-ecs-client/services.tf [hunk 1]
```diff
@@ -125,82 +88,52 @@ resource "aws_ecs_service" "frontend" {
...
-        Effect = "Allow"
-        Principal = {
-          Service = "ecs-tasks.amazonaws.com"
-        }
-      }
-    ]
-  })
-}
-
-resource "aws_iam_role" "public_api-execution-role" {
```
Comment (jjtimmons) [inline modules/hcp-ecs-client/services.tf:159]: as far as I can tell, these are the same as the roles made by default by the `mesh-task` resource: https://github.com/hashicorp/terraform-aws-consul-ecs/blob/v0.3.0/modules/mesh-task/iam.tf#L51

Comment (i0rek) [reply modules/hcp-ecs-client/services.tf:159]: yes, except they have the `local.scope` bit so that you don't run into conflicts. that should stay.

**Review by i0rek — APPROVED** [review]
Looks great overall! Left a comment on why we need our own task roles. `intentions.tf` should be the same for `ecs` and `ec2`. Since you changed one, could you make sure they stay in sync? I wanted to add a command to `make` that compares both, but haven't done that yet.

By bumping the module version, you are preparing for a new release. Before merging this and cutting `0.6.2` I want to take the time to spin up every example.

File: modules/hcp-ecs-client/services.tf (hunk 1 above)
Comment (jjtimmons) [reply modules/hcp-ecs-client/services.tf:159]: ah ack and that makes sense. I saw `family` and thought that'd be enough, but can see how we'd want the random bit as well. Will addback

**Comment by jjtimmons** [comment]
Made a few changes in the last couple changes:

- remove the module bumped (reverted to existing module version)
- applies the nginx ~hack~ approach for EC2 as well
- applied same variable names and ports as for ECS/EKS
- went back to 3 subnets in ECS, only pass the private subnets to the consul cluster
- same `intentions.tf` between the EC2/ECS files

**Comment by jjtimmons** [comment]
Re: above, tested both ECS and EC2 by putting in an order and checking console.

Also logged into the nomad page.
Screen Shot 2022-03-28 at 12 50 04 PM
Screen Shot 2022-03-28 at 1 11 03 PM
Screen Shot 2022-03-28 at 1 10 35 PM

**Pull request merged**